src/integration/java/io/pinecone/integration/dataPlane/QueryErrorPodTest.java

```diff
@@ -23,46 +25,44 @@
 
 public class QueryErrorPodTest {
 
-    private static PineconeConnection connection;
+    private static Pinecone pineconeClient;
+    private static String indexName;
+    private static Index indexClient;
+    private static AsyncIndex asyncIndexClient;
+
     private static final int dimension = 3;
 
     @BeforeAll
     public static void setUp() throws IOException, InterruptedException {
-        connection = createIndexIfNotExistsDataPlane(dimension, IndexModelSpec.SERIALIZED_NAME_POD);
-    }
-
-    @AfterAll
-    public static void cleanUp() {
-        connection.close();
+        AbstractMap.SimpleEntry<String, Pinecone> indexAndClient = createIndexIfNotExistsDataPlane(dimension, IndexModelSpec.SERIALIZED_NAME_POD);
+        indexName = indexAndClient.getKey();
+        pineconeClient = indexAndClient.getValue();
+        indexClient = pineconeClient.createIndexConnection(indexName);
+        asyncIndexClient = pineconeClient.createAsyncIndexConnection(indexName);
```

Comment:
I made this same change consuming the returned tuple from `createIndexIfNotExistsDataPlane` across all of these tests in this folder. It feels a bit easier to reason about, and we test the helper functions on `Pinecone`.

Reply:
Yeah, this is nice. In the near future I can see this moving to an extension that runs at the beginning of the test run and injected for all tests so we only pay the cost once